bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -248,23 +314,69 @@ protected NeoHubReadDcbResponse fromNeoHubReadDcbResponse() {
      * handlers
      */
     private synchronized void lazyPollingSchedulerExecute() {
-        NeoHubInfoResponse infoResponse = fromNeoHubReadInfoResponse();
+        // check which API is supported
+        if (!isApiOnline) {
+            selectApi();
+        }
 
-        if (infoResponse != null) {
-            // determine temperatureUnit
-            NeoHubReadDcbResponse dcbResponse = fromNeoHubReadDcbResponse();
-            Unit<?> temperatureUnit = (dcbResponse != null) ? dcbResponse.getTemperatureUnit() : SIUnits.CELSIUS;
+        NeoHubAbstractDeviceData deviceData = fromNeoHubGetDeviceData();
+        if (deviceData != null) {
+            // check the temperature unit
+            if (systemDataDirty) {
+                NeoHubReadDcbResponse systemData = fromNeoHubReadSystemData();
+                if (systemData != null) {
+                    temperatureUnit = systemData.getTemperatureUnit();
+                    systemDataDirty = false;
+                    logger.info("Setting temperature unit: => {}", temperatureUnit);
```

Comment:
This should be wrapped in a check `if (newTemperatureUnit != oldTemperatureUnit) {}` so this would only be logged on binding start up, of when the user actually changes settings from Fahrenheit to Celsius (which is probably never).

Reply:
Deleted the logging.